File: backend/app/models/enums.py
import enum


class UserRole(str, enum.Enum):
    ADMIN = "admin"
    PHARMACIST = "pharmacist"


class Sex(str, enum.Enum):
    MALE = "male"
    FEMALE = "female"
    OTHER = "other"


class ResidentialSetting(str, enum.Enum):
    HOME = "home"
    RACF = "racf"
    OTHER = "other"


class ReviewType(str, enum.Enum):
    HMR = "hmr"
    RMMR = "rmmr"


class ReviewStatus(str, enum.Enum):
    DRAFT = "draft"
    AWAITING_GP = "awaiting_gp"
    SUBMITTED = "submitted"
    ARCHIVED = "archived"


class ClinicalNoteSection(str, enum.Enum):
    PRESENTING_ISSUES = "presenting_issues"
    MEDICAL_HISTORY = "medical_history"
    ALLERGIES = "allergies"
    ADHERENCE_LIFESTYLE = "adherence_lifestyle"
    PATIENT_GOALS = "patient_goals"


class SuggestionSeverity(str, enum.Enum):
    INFO = "info"
    MODERATE = "moderate"
    HIGH = "high"


class SuggestionCategory(str, enum.Enum):
    INTERACTION = "interaction"
    DOSING = "dosing"
    DEPRESCRIBING = "deprescribing"
    ADHERENCE = "adherence"
    MONITORING = "monitoring"
    OTHER = "other"


class AuditAction(str, enum.Enum):
    CREATED = "created"
    UPDATED = "updated"
    MEDICATIONS_ADDED = "medications_added"
    AI_GENERATED = "ai_generated"
    DRAFT_GENERATED = "draft_generated"
    EDITED = "edited"
    FINALIZED = "finalized"
    REOPENED = "reopened"
    EXPORTED = "exported"

File: frontend/src/types/enums.ts
export enum UserRole {
  ADMIN = 'admin',
  PHARMACIST = 'pharmacist',
}

export enum Sex {
  MALE = 'male',
  FEMALE = 'female',
  OTHER = 'other',
}

export enum ResidentialSetting {
  HOME = 'home',
  RACF = 'racf',
  OTHER = 'other',
}

export enum ReviewType {
  HMR = 'hmr',
  RMMR = 'rmmr',
}

export enum ReviewStatus {
  DRAFT = 'draft',
  AWAITING_GP = 'awaiting_gp',
  SUBMITTED = 'submitted',
  ARCHIVED = 'archived',
}

export enum ClinicalNoteSection {
  PRESENTING_ISSUES = 'presenting_issues',
  MEDICAL_HISTORY = 'medical_history',
  ALLERGIES = 'allergies',
  ADHERENCE_LIFESTYLE = 'adherence_lifestyle',
  PATIENT_GOALS = 'patient_goals',
}

export enum SuggestionSeverity {
  INFO = 'info',
  MODERATE = 'moderate',
  HIGH = 'high',
}

export enum SuggestionCategory {
  INTERACTION = 'interaction',
  DOSING = 'dosing',
  DEPRESCRIBING = 'deprescribing',
  ADHERENCE = 'adherence',
  MONITORING = 'monitoring',
  OTHER = 'other',
}
